main.py

```diff
@@ -162,8 +171,64 @@ def extractImages():
         k = cv2.waitKey(30) & 0xff
         if k == 27:
             break
-        
+
+def extract_foreground(frames, mode):
+    fgmasks = []
+    extractor = ForegroundExtractor()
+    if mode == FG_GRABCUT:
```

Comment:
In the future, if we want to add a new foreground extraction function, we only need to add the new function in ForegroundExtractor and add a new case here